core/src/test/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpServiceTest.java

```diff
@@ -985,6 +1084,64 @@ public void testReturnVoid() throws Exception {
         }
     }
 
+    @Test
+    public void testOriginBlockingTaskType() throws Exception {
+        final HttpClient client = HttpClient.of(server.uri("/"));
+
+        String path = "/12/httpResponse";
+        RequestHeaders headers = RequestHeaders.of(HttpMethod.GET, path);
+        AggregatedHttpResponse res = client.execute(headers).aggregate().join();
+        assertThat(res.status()).isSameAs(HttpStatus.OK);
+        assertThat(blockingCount).hasValue(0);
```

Comment:
I think we can reduce test logic using `@ParameterizedTest` with `@CsvSource` :-) 
```java
@CsvSource({ "/12/httpResponse, 0", "/12/aggregatedHttpResponse, 1", .. })
```